### ISSUE TITLE
Not running in parallel on ray cluster, only on 1 node

### ISSUE THREAD

**Issue by jmakov** [statsmodels/statsmodels#7764]
#### Describe the bug
Looking at https://joblib.readthedocs.io/en/latest/parallel.html `statsmodels` should be able to run on a cluster. Trying the example from the docs (joblib) or ray docs (https://docs.ray.io/en/latest/joblib.html) with `stattools.coint`, it runs in parallel only on the driver node not taking advantage of the free resources on the cluster.

#### Code Sample, a copy-pastable example if possible

```python
import ray
import joblib
from ray.util.joblib import register_ray
from statsmodels.tsa import stattools

ray.init(address='auto', _redis_password='xxx')  # same result if `address` points to the head node
register_ray()

# get your df
with joblib.parallel_backend("ray"):
    stattools.coint(df[df.columns[0]], df[df.columns[1]])  # runs in parallel only on the driver node
```
#### Expected Output
Runs on the whole cluster.

#### Output of ``import statsmodels.api as sm; sm.show_versions()``
```python
INSTALLED VERSIONS
------------------
Python: 3.7.10.final.0
OS: Linux 5.11.0-37-generic #41-Ubuntu SMP Mon Sep 20 16:39:20 UTC 2021 x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

statsmodels
===========

Installed: 0.12.2 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/statsmodels)

Required Dependencies
=====================

cython: 0.29.24 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/Cython)
numpy: 1.20.3 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/numpy)
scipy: 1.5.4 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/scipy)
pandas: 1.3.2 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/pandas)
    dateutil: 2.8.2 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/dateutil)
patsy: 0.5.2 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/patsy)

Optional Dependencies
=====================

matplotlib: 3.4.3 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/matplotlib)
    backend: module://ipympl.backend_nbagg 
cvxopt: Not installed
joblib: 1.0.1 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/joblib)

Developer Tools
================

IPython: 7.27.0 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/IPython)
    jinja2: 3.0.1 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/jinja2)
sphinx: Not installed
    pygments: 2.10.0 (/home/test/miniconda3/envs/puma-lab/lib/python3.7/site-packages/pygments)
pytest: Not installed
virtualenv: Not installed
```

**Comment by bashtage** [comment]
I'm not sure why you think this is a statsmodels bug.  Statsmodels is a mostly python program.  `stattools.coint` is all python except where NumPy is used.  If it isn't running in parallel, this is something to so with how you running the program.


```python
# get your df
with joblib.parallel_backend("ray"):
    stattools.coint(df[df.columns[0]], df[df.columns[1]])  # runs in parallel only on the driver node
```

This code make no sense. There is no use of joblib in coint.  If you want to run `coint` across many datasets, you will need to wire it up yourself.

There is virtually no joblib use in statsmodels:

https://github.com/statsmodels/statsmodels/search?l=Python&q=joblib

**Comment by bashtage** [comment]
You are missing the explicit call to `Parallel`, as in:

```python
Parallel()(delayed(sqrt)(i ** 2) for i in range(10))
```

**Comment by bashtage** [comment]
Scikit-learn explicitly calls `Parallel` in `fit`, e.g., 

https://github.com/scikit-learn/scikit-learn/blob/c588230dad7ce151e6a995e222f25993f405195a/sklearn/model_selection/_search.py#L807

**Comment by jmakov** [comment]
Thanks for the quick response! Since joblib was the only thing in the docs, I thought it's how statmodels achieves parallelism.
